metadata/src/lyrics.rs

```diff
@@ -0,0 +1,79 @@
+use bytes::Bytes;
+
+use librespot_core::{Error, FileId, Session, SpotifyId};
+
+impl LyricsReply {
+    pub async fn get(session: &Session, id: &SpotifyId) -> Result<Self, Error> {
+        let spclient = session.spclient();
+        let lyrics = spclient.get_lyrics(id).await?;
+        LyricsReply::try_from(&lyrics)
+    }
+
+    pub async fn get_for_image(
+        session: &Session,
+        id: &SpotifyId,
+        image_id: &FileId,
+    ) -> Result<Self, Error> {
+        let spclient = session.spclient();
+        let lyrics = spclient.get_lyrics_for_image(id, image_id).await?;
+        LyricsReply::try_from(&lyrics)
+    }
+}
+
+impl TryFrom<&Bytes> for LyricsReply {
+    type Error = Error;
+
+    fn try_from(lyrics: &Bytes) -> Result<Self, Self::Error> {
+        let parsed = serde_json::from_slice(lyrics)?;
+
+        Ok(parsed)
```

Comment:
Probably you could return `serde_json::from_slice(lyrics)` immediately? Maybe with `map_err(Into::into)`?